### PR TITLE
[Fix] Deflake score engine determinism tests

### LINKED ISSUE #30523: [Bug] Flaky TestSeqClsScoring score determinism in 1-GPU CI
[by Codex]

## Summary

`TestSeqClsScoring.test_score_deterministic` can return materially different scores for two identical consecutive requests, causing the 1-GPU base CI shard to fail after its retry.

## Failure

- CI job: https://github.com/sgl-project/sglang/actions/runs/28922429584/job/85807458706
- Test: `test/registered/prefill_only/test_score_engine.py::TestSeqClsScoring::test_score_deterministic`
- Model setup: Qwen3-0.6B backbone overridden to `Qwen3ForSequenceClassification` with a randomly initialized two-label classification head.

The test invokes `Engine.score()` twice with the same query and items and compares all returned probabilities to one decimal place. The failing pair was:

```text
0.953125 != 0.90234375 within 1 places
(0.05078125 difference)
```

The same pair failed again when `CustomTestCase` retried the method, after which the test exited with `retry() exceed maximum number of retries`.

The preceding `TestCausalLMScoring.test_score_deterministic` also observed a smaller first-attempt difference (`1.2238e-05` at five decimal places), but passed its retry.

## Scope assessment

This was observed on PR #29929, which only changes the FlashInfer expert-parallel A2A token dispatcher and adds a 4-GPU GLM-5.2 NVFP4 test. The failing job is a 1-GPU sequence-classification scoring path and does not enable EP or FlashInfer A2A. The PR changes only:

- `python/sglang/srt/layers/moe/token_dispatcher/flashinfer.py`
- `test/registered/4-gpu-models/test_glm52_nvfp4.py`

Therefore the failure does not appear to be caused by PR #29929.

As of current `main` (`3d96bb9721`), there are no subsequent changes to `test_score_engine.py` and no open issue or PR specifically tracking/fixing this sequence-classification determinism failure.

## Suggested investigation

- Determine why two identical requests in the same engine instance can observe different sequence-classification outputs, including whether CUDA-graph replay or shared output buffers are involved.
- Since the classification head is randomly initialized, make its initialization deterministic if the test depends on a stable numerical tolerance.
- Revisit the tolerance only after confirming that the difference is expected numerical variation rather than a stale-buffer or replay correctness issue.

### PR DESCRIPTION
**Pull request by hnyls2002**
## Summary
Fix flaky `test_score_deterministic` in `test_score_engine.py` (fixes #30523).

- `TestCausalLMScoring`: flush cache between the two calls so both take the fresh-prefill path (previously the second call hit the radix cache prefix, a different kernel path with bf16-level divergence), and replace `places=5` with a relative tolerance
- `TestSeqClsScoring`: pin `random_seed=42` so the randomly initialised classification head is reproducible across CI runs, and loosen `places=1` to `delta=0.2` on the raw bf16 logits

The determinism checks still guard their real failure mode (stale output buffers / cross-request contamination, which produce order-of-magnitude differences); the looser tolerances only absorb benign bf16 batch-composition noise.

## Verification
`/rerun-test test_score_engine.py` dispatched 10x on this PR, 10/10 test pass:
[1](https://github.com/sgl-project/sglang/actions/runs/29535555147)
[2](https://github.com/sgl-project/sglang/actions/runs/29536254922)
[3](https://github.com/sgl-project/sglang/actions/runs/29537336952)
[4](https://github.com/sgl-project/sglang/actions/runs/29537522944)
[5](https://github.com/sgl-project/sglang/actions/runs/29538081083)
[6](https://github.com/sgl-project/sglang/actions/runs/29538288700)
[7](https://github.com/sgl-project/sglang/actions/runs/29538505019)
[8](https://github.com/sgl-project/sglang/actions/runs/29538754048)
[9](https://github.com/sgl-project/sglang/actions/runs/29538964243)
[10](https://github.com/sgl-project/sglang/actions/runs/29539162206)

Run 10 shows red only because its `write-back-result` step hit a GitHub API 503 outage; the test job itself passed.

---
### CI States

Latest PR Test (Base): :x: [Run #29540203657](https://github.com/sgl-project/sglang/actions/runs/29540203657)
Latest PR Test (Extra): :x: [Run #29540203497](https://github.com/sgl-project/sglang/actions/runs/29540203497)

### CODE REVIEW

**Comment by hnyls2002** [comment]
/rerun-test test_score_engine.py
(10 tries)
